src/ECharts/Legend.purs

```diff
@@ -0,0 +1,91 @@
+module ECharts.Legend where
+
+import Data.Maybe
+import Data.StrMap 
+import Data.Argonaut.Core
+import Data.Argonaut.Encode
+import Data.Argonaut.Combinators
+
+import ECharts.Color
+import ECharts.Common
+import ECharts.Coords
+import ECharts.Style.Text
+import ECharts.Formatter
+
+data LegendItem = LegendString String |
+  LegendItem {
+    "name" :: Maybe String,
```

Comment:
Are all 3 of these really optional? It seems like the library couldn't handle that case (but maybe it can).